datafusion/substrait/tests/cases/roundtrip_logical_plan.rs

```diff
@@ -571,6 +571,21 @@ async fn roundtrip_self_implicit_cross_join() -> Result<()> {
     roundtrip("SELECT left.a left_a, left.b, right.a right_a, right.c FROM data AS left, data AS right").await
 }
 
+#[tokio::test]
+async fn self_join_introduces_aliases() -> Result<()> {
```

Comment:
This is adding back [this test](https://github.com/apache/datafusion/pull/11049/files#diff-26c40258720a61cb066cd14651b9b3617557cf7394838b3bad6b4e45718b3122L645), right? I seem to have argued back then that it is unnecessary given the roundtrip_self_join test.